packages/sdk/src/FeeService.ts

```diff
@@ -0,0 +1,168 @@
+  m: number;
+  /**
+   * @description high
+   */
+  h: number;
+  /**
+   * @description very high
+   */
+  vh: number;
+}
+
+interface MarketReferenceFee {
+  claim: number;
+  jup: Fee;
+  jup2: Fee;
+  loAndDCA: number;
+  referral: number;
+  perps: Fee;
+  swapFee: number;
+  lastUpdatedAt: number;
+}
+
+interface GetOptimalComputeUnitLimitAndPricePayload {
+  instructions: TransactionInstruction[];
+  payer: PublicKey;
+  lookupTables: AddressLookupTableAccount[];
+}
+
+interface GetOptimalComputeUnitLimitAndPriceResponse {
+  units: number;
+  microLamports: number;
+}
+
+interface FeeService {
+  getOptimalComputeUnitLimitAndPrice: (
+    payload: GetOptimalComputeUnitLimitAndPricePayload,
+  ) => Promise<GetOptimalComputeUnitLimitAndPriceResponse>;
+}
+
+class FeeServiceImpl implements FeeService {
+  // --------------------
+  // Properties
+  // --------------------
+  private connection: Connection;
+
+  constructor() {
+    this.connection = new Connection(RPC_URL);
+  }
+
+  // --------------------
+  // API
+  // --------------------
+  private getMarketReferenceFee = async (): Promise<MarketReferenceFee> => {
+    // TODO: To update the URL to `https://cache.jup.ag/reference-fees` once dependencies PR merged
+    const data = (
+      await fetch("https://cache.jup.ag/jup-claim-reference-fees")
+    ).json() as unknown as MarketReferenceFee;
+    return data;
+  };
+
+  // --------------------
+  // Helper methods
+  // --------------------
+
+  /**
+   *
+   * Code snippets from the Solana documentation
+   * @see https://solana.com/developers/guides/advanced/how-to-request-optimal-compute#how-to-request-compute-budget
+   */
+  private getSimulationUnits = async (
+    payload: GetOptimalComputeUnitLimitAndPricePayload,
+  ) => {
+    const { instructions, payer, lookupTables } = payload;
+
+    const testInstructions = [
+      ComputeBudgetProgram.setComputeUnitLimit({ units: 1_400_000 }),
+      ...instructions,
+    ];
+
+    const testVersionedTxn = new VersionedTransaction(
+      new TransactionMessage({
+        instructions: testInstructions,
+        payerKey: payer,
+        recentBlockhash: PublicKey.default.toString(),
+      }).compileToV0Message(lookupTables),
+    );
+
+    const simulation = await this.connection.simulateTransaction(
+      testVersionedTxn,
+      {
+        replaceRecentBlockhash: true,
+        sigVerify: false,
+      },
+    );
+    if (simulation.value.err) {
+      return undefined;
+    }
+    return simulation.value.unitsConsumed;
+  };
+
+  private addMarginErrorForComputeUnitLimit = (units: number, margin: number) =>
+    Math.floor(units * margin);
+
+  private getReferralReferenceFeeInMicroLamports = async (
+    computeUnitLimit: number,
+  ) => {
+    const marketReferenceFeeInLamports = await this.getMarketReferenceFee();
+
+    const referralReferenceFeeInMicroLamports =
+      this.computePriceMicroLamportsFromFeeLamports(
+        marketReferenceFeeInLamports.referral,
```

Comment:
fee is already in `microLamports`, don't have to do `computePriceMicroLamportsFromFeeLamports` anymore. can just set IIRC.

Reply:
Hmm, because currently `monorepo` is converting it using the same formula before using. Does this mean I should update the `monorepo/jupiter-ui` as well

https://github.com/mercurial-finance/monorepo/blob/501f5d77ed91d24f60c45b593a45ac9db4cb627a/apps/jupiter-ui/src/contexts/PrioritizationFeeContext/index.tsx#L120

Reply:
Removed this!